lib/main/atom/commands/findReferences.tsx

```diff
@@ -14,44 +15,51 @@ addCommand("atom-text-editor", "typescript:find-references", deps => ({
 
     const client = await deps.getClient(location.file)
     const result = await client.execute("references", location)
-    const refs = await Promise.all(
-      result.body!.refs.map(async ref => {
-        const fileContents = (
-          await new Promise<string>((resolve, reject) =>
-            fs.readFile(ref.file, (error, data) => {
-              if (error) reject(error)
-              else resolve(data.toString("utf-8"))
-            }),
-          )
-        ).split(/\r?\n/g)
-        const context =
-          ref.contextStart !== undefined && ref.contextEnd !== undefined
-            ? fileContents.slice(ref.contextStart.line - 1, ref.contextEnd.line)
-            : fileContents
-        const fileHlText = await highlight(context.join("\n"), "source.tsx")
-        // tslint:disable-next-line: strict-boolean-expressions
-        const lineText = fileHlText[ref.start.line - (ref.contextStart?.line || 1)]
-        return {...ref, hlText: lineText}
-      }),
-    )
-    console.log(refs)
-
-    const res = await selectListView({
-      items: refs,
-      itemTemplate: (item, ctx) => {
-        return (
-          <li>
-            <HighlightComponent
-              label={atom.project.relativize(item.file)}
-              query={ctx.getFilterQuery()}
-            />
-            <div className="pull-right">line: {item.start.line}</div>
-            <TsView highlightedText={item.hlText} />
-          </li>
-        )
-      },
-      itemFilterKey: "file",
-    })
-    if (res) await deps.histGoForward(editor, res)
+    await handleFindReferencesResult(result, editor, deps.histGoForward)
   },
 }))
+
+export async function handleFindReferencesResult(
+  result: protocol.ReferencesResponse,
+  editor: TextEditor,
+  histGoForward: Dependencies["histGoForward"],
+): Promise<void> {
+  const refs = await Promise.all(
```

Comment:
While you're at it, could you also get rid of this `await`? It's a holdover from way back, and it's really not needed here. Removing it will make the 'find-references' feel way more responsive. You shouldn't need to change anything else for it to work.

Reply:
Ok. I did remove that `await`.